gpkit/constraints/base.py

```diff
@@ -0,0 +1,54 @@
+"Implements BaseConstraint"
+from collections import defaultdict, Iterable
+from .set import ConstraintSet
+from ..varkey import VarKey
+from ..nomials import Variable, Signomial
+from .. import SignomialsEnabled
+
+
+class ConstraintBase(ConstraintSet):
+    "A ConstraintSet for making named models with setup methods"
```

Comment:
Create and solve programs, mostly, plus `__and__` and `__or__` for backwards compatibility.